src/test_lib_json/main.cpp

```diff
@@ -3903,6 +3903,37 @@ JSONTEST_FIXTURE_LOCAL(FuzzTest, fuzzDoesntCrash) {
                              example.size()));
 }
 
+struct GetLocationLineAndColumnTest : JsonTest::TestCase {
+  void testLineAndColumn(const std::string& doc, ptrdiff_t location, int row,
+                         int column) {
+    Json::LineAndColumn actualLocation = Json::getLocationLineAndColumn(
+        doc.data(), doc.data() + doc.length(), location);
+    JSONTEST_ASSERT_EQUAL(row, actualLocation.line);
+    JSONTEST_ASSERT_EQUAL(column, actualLocation.column);
+  }
+};
+
+JSONTEST_FIXTURE_LOCAL(GetLocationLineAndColumnTest, test) {
+  const std::string example = "line 1\nline 2\r\nline 3\rline 4";
+  size_t found_l = -1;
+  size_t found_i = -1;
+  size_t found_n = -1;
+  size_t found_e = -1;
+  size_t found_space = -1;
+  for (int row = 1; row <= 4; ++row) {
+    found_l = example.find('l', found_l + 1);
+    found_i = example.find('i', found_i + 1);
+    found_n = example.find('n', found_n + 1);
+    found_e = example.find('e', found_e + 1);
+    found_space = example.find(' ', found_space + 1);
+    testLineAndColumn(example, found_l, row, 1);
+    testLineAndColumn(example, found_i, row, 2);
+    testLineAndColumn(example, found_n, row, 3);
+    testLineAndColumn(example, found_e, row, 4);
+    testLineAndColumn(example, found_space, row, 5);
```

Comment:
I don't think the 3rd, 4th, and 5th finds tell you anything the first 2 didn't already tell you.